### PR TITLE
Fix node/vertex confusion

### PR DESCRIPTION
**Pull request by johnnonweiler**
For every triangle (facet) vertex, a node and meshset were being added to MOAB.  With this change, a meshset is only added for vertices which exist in the original geometry, and "nodes" continue to be added for every triangle (facet) vertex.

Also, a new cylinder faceting test is added to test this change, since in the other test there are no nodes which are not vertices, so this change should make no difference there.

This also fixes a bug in raise_moab_error(), where unexpected rvals were causing a dodgy memory access.

### CODE REVIEW

File: src/brep_faceter.cc [hunk 1]
```diff
@@ -189,6 +191,27 @@ void facet_all_volumes(const TopTools_HSequenceOfShape &shape_list,
 
           edge_data edges = make_edge_facets(currentEdge, data);
           mbtool.build_curve(curve, edges, f_vertex_map);
+
+          // add vertices to edges
+          TopTools_IndexedMapOfShape vertices;
+          TopExp::MapShapes(currentEdge, TopAbs_VERTEX, vertices);
+          for (int j = 1; j < vertices.Extent(); j++) {
```
Comment (smason) [inline src/brep_faceter.cc]: is this deliberately ignoring the last element?

OCCT indices tend to go over `[1,n]`, not the normal C style `[0,n)`.

In this case see [NCollection_IndexedMap::FindKey()](https://git.dev.opencascade.org/gitweb/?p=occt.git;a=blob;f=src/NCollection/NCollection_IndexedMap.hxx;h=875068d83c5c3bda49f636134f4017251148e323;hb=HEAD#l372)

Comment (johnnonweiler) [reply src/brep_faceter.cc]: Thank you @smason !  That was the problem.  Good spot.

File: src/test/test_cylinder_faceting.cc [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  materials_map[uniqueID] = "testmat";
+
+  sew_and_facet(shape, facet_tol, mbtool, materials_map);
+
+  std::vector<moab::EntityHandle> triangles = mbtool.get_entities_by_dimension(0, 2, true);
+
+  // This is a check if something has changed - I haven't independantly checked this value:
+  CHECK(triangles.size() == 322);
+
+  // 2 vertices + 3 edges + 3 faces + 1 volume + 1 material group = 10
```
Comment (johnnonweiler) [inline src/test/test_cylinder_faceting.cc]: Although this cylinder faceting test is passing, I'm far from convinced that there are sufficient checks, or that the output file is correct.  The comment `// 2 vertices + 3 edges + 3 faces + 1 volume + 1 material group = 10` appears to be wrong - I think there are other reasons for the 10 meshsets (which include the gather set).

**Comment by johnnonweiler** [comment]
I have changed the tests so that they both pass.  (This involved a hack to make the faceter_test output match the test output from Cubit.  That hack didn't affect the cylinder test, which had previously been passing, so I don't know what changed there.)

I have added a "todo" comment, about the fact the cylinder test needs more checks (as I mentioned in my comment above).

**Comment by johnnonweiler** [comment]
@smason I am going to merge this onto the other branch now, so that my upstream PR is updated.  Any further comments could still be added here (or recorded elsewhere).

**Pull request merged**